Adamant/Services/HealthCheckWrapper.swift

```diff
@@ -94,17 +94,45 @@ class HealthCheckWrapper<Service, Error: HealthCheckableError> {
             .store(in: &subscriptions)
     }
     
+    func waitingRequest<Output>(
+        _ requestAction: @Sendable @escaping (Service, Node) async -> Result<Output, Error>
+    ) async -> Result<Output, Error> {
+        guard allowedNodes.isEmpty else {
+            return await handleRequest(requestAction: requestAction)
+        }
+        
+        return await withCheckedContinuation { [weak self] continuation in
+            guard let self = self else {
+                return
+            }
+            $allowedNodes
+                .filter { !$0.isEmpty }
+                .prefix(1)
+                .sink { [weak self] _ in
+                    guard let self = self else {
+                        continuation.resume(returning: .failure(.noEndpointsError(coin: NodeGroup.adm.name)))
```

Comment:
This health check is a universal class used for all the blockchains. Don't use the ADM related error